### ISSUE TITLE
Incorrect startup sequence of plugins

### ISSUE THREAD

**Issue by exceet-pk** [opnsense/core#3724]
**Describe the bug**
We're using OPNSense OPNsense 19.7.4_1 (and previous versions) on ten firewall systems since the beginning of 2019 with following plugins:

- os-frr
- os-haproxy

The RIP implementation from the os-frr plugin is used to distribute routes in our infrastructure and HAP from the os-haproxy plugin is used as load balancer for our various services.

Since we're using OPNSense we've had the problem, that during the startup sequence HAP starts before RIP. This causes the problem that HAP cannot find its backend systems and we have to wait about 30 minutes until all connection attempts to them are timed out. We can't even abort the initialization of HAP with (ctrl)+(c), because then RIP will also not be started. 

We would be really appreciated if you can adjust the startup sequence in a proper way. 

**To Reproduce**
1. Configure and activate RIP
2. Configure HAP with a backend service only available after RIP distributed the corresponding route
3. Reboot OPNSense

**Expected behavior**
We're expecting that RIP and other dynamic routing protocols will be initialized before other plugins like HAP will be started. 

**Environment**
Software version used and hardware type if relevant.
e.g.:

OPNsense 19.7.4_1-amd64
FreeBSD 11.2-RELEASE-p14-HBSD
OpenSSL 1.0.2s 28 May 2019

os-acme-client (installed) = 1.25
os-dyndns (installed) = 1.17
os-frr (installed) = 1.11_3
os-haproxy (installed) = 2.18
os-vmware (installed) = 1.5

**Comment by fichtner** [comment]
From what I can tell the ports are set up as follows during startup:

FRR: # REQUIRE: netif routing  
HAProxy: # REQUIRE: DAEMON LOGIN

LOGIN and DAEMON require NETWORKING which netif and routing are a part of... which means HAProxy starts after FRR, which means this is a different type of race condition.

You can likely confirm this by:

# sh -c "rcorder -s nostart -s firsttime /etc/rc.d/* /usr/local/etc/rc.d/* 2> /dev/null | grep -e ffr -e haproxy"

It should print the correct order.

Now, *if* one of the services fails to start on their boot spot that is another matter to look into. I know for FRR we try to restart later just to be sure.

What would be needed to debug this is the full console output of the boot sequence.


Cheers,
Franco

**Comment by fraenki** [comment]
> This causes the problem that HAP cannot find its backend systems and we have to wait about 30 minutes until all connection attempts to them are timed out.

This part sounds interesting to me. What makes you so sure that the 30 minute timeout is due to HAProxy waiting for backend systems? Usually HAProxy's startup is not delayed by backend, servers, etc.

**Comment by exceet-pk** [comment]
Please excuse my late reply, but currently, I'm very busy with other operational stuff. 

First of all, I'd like to clarify that the problem we have is a bit different than I described in the first place. It's the same setup as described above, but it's important to know that our DNS server is also only reachable after ffr got initialized and that we configured the HAP's "real servers" with their hostnames and not their IP addresses. 

To give you some more information I ran Franco's command to confirm the startup sequence and pasted the complete console output of the boot sequence. 

`root@anonymized_hostname:~ # sh -c "rcorder -s nostart -s firsttime /etc/rc.d/* /usr/local/etc/rc.d/* 2> /dev/null | grep -e frr -e haproxy"
/usr/local/etc/rc.d/frr
/usr/local/etc/rc.d/watchfrr
/usr/local/etc/rc.d/haproxy
`

`Copyright (c) 2013-2018 The HardenedBSD Project.
Copyright (c) 1992-2018 The FreeBSD Project.
Copyright (c) 1979, 1980, 1983, 1986, 1988, 1989, 1991, 1992, 1993, 1994
	The Regents of the University of California. All rights reserved.
FreeBSD is a registered trademark of The FreeBSD Foundation.
FreeBSD 11.2-RELEASE-p14-HBSD  07680caafe9(stable/19.7) amd64
FreeBSD clang version 6.0.0 (tags/RELEASE_600/final 326565) (based on LLVM 6.0.0)
VT(vga): text 80x25
HardenedBSD: initialize and check features (__HardenedBSD_version 1100056 __FreeBSD_version 1102000).
CPU: Intel(R) Xeon(R) CPU E5-2640 v4 @ 2.40GHz (2400.00-MHz K8-class CPU)
  Origin="GenuineIntel"  Id=0x406f0  Family=0x6  Model=0x4f  Stepping=0
  Features=0xf83fbff<FPU,VME,DE,PSE,TSC,MSR,PAE,MCE,CX8,APIC,SEP,MTRR,PGE,MCA,CMOV,PAT,PSE36,MMX,FXSR,SSE,SSE2,SS>
  Features2=0xfffa3203<SSE3,PCLMULQDQ,SSSE3,FMA,CX16,PCID,SSE4.1,SSE4.2,x2APIC,MOVBE,POPCNT,TSCDLT,AESNI,XSAVE,OSXSAVE,AVX,F16C,RDRAND,HV>
  AMD Features=0x2c100800<SYSCALL,NX,Page1GB,RDTSCP,LM>
  AMD Features2=0x121<LAHF,ABM,Prefetch>
  Structured Extended Features=0x1c2fbb<FSGSBASE,TSCADJ,BMI1,HLE,AVX2,SMEP,BMI2,ERMS,INVPCID,RTM,NFPUSG,RDSEED,ADX,SMAP>
  Structured Extended Features3=0xbc000000<IBPB,STIBP,ARCH_CAP,SSBD>
  XSAVE Features=0x1<XSAVEOPT>
  IA32_ARCH_CAPS=0xc
  TSC: P-state invariant
Hypervisor: Origin = "VMwareVMware"
real memory  = 4294967296 (4096 MB)
avail memory = 4083200000 (3894 MB)
Event timer "LAPIC" quality 600
ACPI APIC Table: <PTLTD  	 APIC  >
FreeBSD/SMP: Multiprocessor System Detected: 4 CPUs
FreeBSD/SMP: 4 package(s)
MADT: Forcing active-low polarity and level trigger for SCI
ioapic0 <Version 1.1> irqs 0-23 on motherboard
SMP: AP CPU #2 Launched!
SMP: AP CPU #3 Launched!
SMP: AP CPU #1 Launched!
random: entropy device external interface
wlan: mac acl policy registered
000.000017 [4213] netmap_init               netmap: loaded module
module_register_init: MOD_LOAD (vesa, 0xffffffff8114ade0, 0) error 19
random: registering fast source Intel Secure Key RNG
random: fast provider: "Intel Secure Key RNG"
kbd1 at kbdmux0
nexus0
vtvga0: <VT VGA driver> on motherboard
cryptosoft0: <software crypto> on motherboard
acpi0: <INTEL 440BX> on motherboard
acpi0: Power Button (fixed)
hpet0: <High Precision Event Timer> iomem 0xfed00000-0xfed003ff on acpi0
Timecounter "HPET" frequency 14318180 Hz quality 950
cpu0: <ACPI CPU> numa-domain 0 on acpi0
cpu1: <ACPI CPU> numa-domain 0 on acpi0
cpu2: <ACPI CPU> numa-domain 0 on acpi0
cpu3: <ACPI CPU> numa-domain 0 on acpi0
attimer0: <AT timer> port 0x40-0x43 irq 0 on acpi0
Timecounter "i8254" frequency 1193182 Hz quality 0
Event timer "i8254" frequency 1193182 Hz quality 100
atrtc0: <AT realtime clock> port 0x70-0x71 irq 8 on acpi0
atrtc0: registered as a time-of-day clock, resolution 1.000000s
Event timer "RTC" frequency 32768 Hz quality 0
Timecounter "ACPI-fast" frequency 3579545 Hz quality 900
acpi_timer0: <24-bit timer at 3.579545MHz> port 0x1008-0x100b on acpi0
pcib0: <ACPI Host-PCI bridge> port 0xcf8-0xcff on acpi0
pci0: <ACPI PCI bus> on pcib0
pcib1: <ACPI PCI-PCI bridge> at device 1.0 on pci0
pci1: <ACPI PCI bus> on pcib1
isab0: <PCI-ISA bridge> at device 7.0 on pci0
isa0: <ISA bus> on isab0
atapci0: <Intel PIIX4 UDMA33 controller> port 0x1f0-0x1f7,0x3f6,0x170-0x177,0x376,0x1060-0x106f at device 7.1 on pci0
ata0: <ATA channel> at channel 0 on atapci0
ata1: <ATA channel> at channel 1 on atapci0
pci0: <bridge> at device 7.3 (no driver attached)
vgapci0: <VGA-compatible display> port 0x1070-0x107f mem 0xe8000000-0xefffffff,0xfe000000-0xfe7fffff irq 16 at device 15.0 on pci0
vgapci0: Boot video device
mpt0: <LSILogic 1030 Ultra4 Adapter> port 0x1400-0x14ff mem 0xfeba0000-0xfebbffff,0xfebc0000-0xfebdffff irq 17 at device 16.0 on pci0
mpt0: MPI Version=1.2.0.0
pcib2: <ACPI PCI-PCI bridge> at device 17.0 on pci0
pci2: <ACPI PCI bus> on pcib2
em0: <Intel(R) PRO/1000 Legacy Network Connection 1.1.0> port 0x2000-0x203f mem 0xfd580000-0xfd59ffff,0xfdff0000-0xfdffffff irq 18 at device 0.0 on pci2
em0: Ethernet address: 00:50:56:86:40:b9
em0: link state changed to UP
em0: netmap queues/slots: TX 1/256, RX 1/256
em1: <Intel(R) PRO/1000 Legacy Network Connection 1.1.0> port 0x2040-0x207f mem 0xfd560000-0xfd57ffff,0xfdfe0000-0xfdfeffff irq 16 at device 2.0 on pci2
em1: Ethernet address: 00:50:56:86:cb:0c
em1: link state changed to UP
em1: netmap queues/slots: TX 1/256, RX 1/256
em2: <Intel(R) PRO/1000 Legacy Network Connection 1.1.0> port 0x2080-0x20bf mem 0xfd540000-0xfd55ffff,0xfdfd0000-0xfdfdffff irq 17 at device 3.0 on pci2
em2: Ethernet address: 00:50:56:86:f1:b2
em2: link state changed to UP
em2: netmap queues/slots: TX 1/256, RX 1/256
em3: <Intel(R) PRO/1000 Legacy Network Connection 1.1.0> port 0x20c0-0x20ff mem 0xfd520000-0xfd53ffff,0xfdfc0000-0xfdfcffff irq 18 at device 4.0 on pci2
em3: Ethernet address: 00:50:56:86:45:43
em3: link state changed to UP
em3: netmap queues/slots: TX 1/256, RX 1/256
em4: <Intel(R) PRO/1000 Legacy Network Connection 1.1.0> port 0x2400-0x243f mem 0xfd500000-0xfd51ffff,0xfdfb0000-0xfdfbffff irq 19 at device 5.0 on pci2
em4: Ethernet address: 00:50:56:86:6e:04
em4: link state changed to UP
em4: netmap queues/slots: TX 1/256, RX 1/256
pcib3: <ACPI PCI-PCI bridge> at device 21.0 on pci0
pcib3: [GIANT-LOCKED]
pcib4: <ACPI PCI-PCI bridge> at device 21.1 on pci0
pcib4: [GIANT-LOCKED]
pcib5: <ACPI PCI-PCI bridge> at device 21.2 on pci0
pcib5: [GIANT-LOCKED]
pcib6: <ACPI PCI-PCI bridge> at device 21.3 on pci0
pcib6: [GIANT-LOCKED]
pcib7: <ACPI PCI-PCI bridge> at device 21.4 on pci0
pcib7: [GIANT-LOCKED]
pcib8: <ACPI PCI-PCI bridge> at device 21.5 on pci0
pcib8: [GIANT-LOCKED]
pcib9: <ACPI PCI-PCI bridge> at device 21.6 on pci0
pcib9: [GIANT-LOCKED]
pcib10: <ACPI PCI-PCI bridge> at device 21.7 on pci0
pcib10: [GIANT-LOCKED]
pcib11: <ACPI PCI-PCI bridge> at device 22.0 on pci0
pcib11: [GIANT-LOCKED]
pcib12: <ACPI PCI-PCI bridge> at device 22.1 on pci0
pcib12: [GIANT-LOCKED]
pcib13: <ACPI PCI-PCI bridge> at device 22.2 on pci0
pcib13: [GIANT-LOCKED]
pcib14: <ACPI PCI-PCI bridge> at device 22.3 on pci0
pcib14: [GIANT-LOCKED]
pcib15: <ACPI PCI-PCI bridge> at device 22.4 on pci0
pcib15: [GIANT-LOCKED]
pcib16: <ACPI PCI-PCI bridge> at device 22.5 on pci0
pcib16: [GIANT-LOCKED]
pcib17: <ACPI PCI-PCI bridge> at device 22.6 on pci0
pcib17: [GIANT-LOCKED]
pcib18: <ACPI PCI-PCI bridge> at device 22.7 on pci0
pcib18: [GIANT-LOCKED]
pcib19: <ACPI PCI-PCI bridge> at device 23.0 on pci0
pcib19: [GIANT-LOCKED]
pcib20: <ACPI PCI-PCI bridge> at device 23.1 on pci0
pcib20: [GIANT-LOCKED]
pcib21: <ACPI PCI-PCI bridge> at device 23.2 on pci0
pcib21: [GIANT-LOCKED]
pcib22: <ACPI PCI-PCI bridge> at device 23.3 on pci0
pcib22: [GIANT-LOCKED]
pcib23: <ACPI PCI-PCI bridge> at device 23.4 on pci0
pcib23: [GIANT-LOCKED]
pcib24: <ACPI PCI-PCI bridge> at device 23.5 on pci0
pcib24: [GIANT-LOCKED]
pcib25: <ACPI PCI-PCI bridge> at device 23.6 on pci0
pcib25: [GIANT-LOCKED]
pcib26: <ACPI PCI-PCI bridge> at device 23.7 on pci0
pcib26: [GIANT-LOCKED]
pcib27: <ACPI PCI-PCI bridge> at device 24.0 on pci0
pcib27: [GIANT-LOCKED]
pcib28: <ACPI PCI-PCI bridge> at device 24.1 on pci0
pcib28: [GIANT-LOCKED]
pcib29: <ACPI PCI-PCI bridge> at device 24.2 on pci0
pcib29: [GIANT-LOCKED]
pcib30: <ACPI PCI-PCI bridge> at device 24.3 on pci0
pcib30: [GIANT-LOCKED]
pcib31: <ACPI PCI-PCI bridge> at device 24.4 on pci0
pcib31: [GIANT-LOCKED]
pcib32: <ACPI PCI-PCI bridge> at device 24.5 on pci0
pcib32: [GIANT-LOCKED]
pcib33: <ACPI PCI-PCI bridge> at device 24.6 on pci0
pcib33: [GIANT-LOCKED]
pcib34: <ACPI PCI-PCI bridge> at device 24.7 on pci0
pcib34: [GIANT-LOCKED]
acpi_acad0: <AC Adapter> on acpi0
atkbdc0: <Keyboard controller (i8042)> port 0x60,0x64 irq 1 on acpi0
atkbd0: <AT Keyboard> irq 1 on atkbdc0
kbd0 at atkbd0
atkbd0: [GIANT-LOCKED]
psm0: <PS/2 Mouse> irq 12 on atkbdc0
psm0: [GIANT-LOCKED]
psm0: model IntelliMouse, device ID 3
acpi_syscontainer0: <System Container> on acpi0
orm0: <ISA Option ROMs> at iomem 0xc0000-0xc7fff,0xc8000-0xc9fff,0xca000-0xcafff,0xcb000-0xcbfff,0xcc000-0xccfff,0xcd000-0xcdfff,0xce000-0xcefff,0xdc000-0xdffff,0xe0000-0xe7fff on isa0
vga0: <Generic ISA VGA> at port 0x3c0-0x3df iomem 0xa0000-0xbffff on isa0
ppc0: cannot reserve I/O port range
Timecounters tick every 10.000 msec
usb_needs_explore_all: no devclass
da0 at mpt0 bus 0 scbus2 target 0 lun 0
da0: <VMware Virtual disk 2.0> Fixed Direct Access SPC-4 SCSI device
da0: 320.000MB/s transfers (160.000MHz, offset 127, 16bit)
da0: Command Queueing enabled
da0: 20480MB (41943040 512 byte sectors)
da0: quirks=0x140<RETRY_BUSY,STRICT_UNMAP>
random: unblocking device.
Trying to mount root from ufs:/dev/gpt/rootfs [rw,noatime]...
Mounting filesystems...
tunefs: soft updates remains unchanged as enabled
tunefs: file system reloaded
camcontrol: ATA ATAPI_IDENTIFY via pass_16 failed
** /dev/gpt/rootfs
FILE SYSTEM CLEAN; SKIPPING CHECKS
clean, 4413382 free (10102 frags, 550410 blocks, 0.2% fragmentation)
** /dev/gpt/rootfs
FILE SYSTEM CLEAN; SKIPPING CHECKS
clean, 4413382 free (10102 frags, 550410 blocks, 0.2% fragmentation)
Setting hostuuid: 4dfc0642-1193-f46a-edf0-aa064c06030a.
Setting hostid: 0xfbc189e2.
Configuring vt: keymap blanktime.
Configuring crash dump device: /dev/null
.ELF ldconfig path: /lib /usr/lib /usr/lib/compat /usr/local/lib /usr/local/lib/ipsec /usr/local/lib/perl5/5.30/mach/CORE
32-bit compatibility ldconfig path:
done.
>>> Invoking early script 'update'
>>> Invoking early script 'configd'
Starting configd.
>>> Invoking early script 'templates'
Generating configuration: OK
>>> Invoking early script 'backup'
>>> Invoking backup script 'captiveportal'
>>> Invoking backup script 'dhcpleases'
>>> Invoking backup script 'duid'
>>> Invoking backup script 'netflow'
>>> Invoking backup script 'rrd'
>>> Invoking early script 'vmware'
Loading vmmemctl kernel module: 
VMware memory control driver initialized
done.
Loading vmxnet kernel module: done.
Loading vmblock kernel module: done.
>>> Invoking early script '98_carp_maintenance'
>>> Invoking early script 'carp_service_status'
OK
Launching the init system...done.
Initializing...........done.
Starting device manager...done.
Configuring login behaviour...done.
Configuring loopback interface...done.
Configuring kernel modules...
aesni0: <AES-CBC,AES-XTS,AES-GCM,AES-ICM> on motherboard
done.
Setting up extended sysctls...sysctl: oid 'kern.randompid' is read only
sysctl: oid 'vm.pmap.pti' is a read only tunable
sysctl: Tunable values are set in /boot/loader.conf
sysctl: oid 'kern.randompid' is read only
sysctl: oid 'vm.pmap.pti' is a read only tunable
sysctl: Tunable values are set in /boot/loader.conf
done.
Setting timezone...done.
Writing firmware setting...done.
Writing trust files...done.
Setting hostname: $anonymized_hostname$
Generating /etc/hosts...done.
Configuring system logging...done.
Configuring loopback interface...done.
Creating wireless clone interfaces...done.
Configuring dmz interface...done.
Configuring signamus_prod_backend interface...done.
Configuring signamus_prod_database interface...done.
Configuring signamus_prod_frontend interface...done.
Configuring transfer interface...done.
Creating IPsec VTI instances...done.
Generating /etc/resolv.conf...done.
Configuring firewall.......done.
Starting PFLOG...done.
pflog0: promiscuous mode enabled
Configuring OpenSSH...done.
Starting web GUI...done.
Configuring CRON...done.
Setting up routes...done.
Generating /etc/hosts...done.
Starting DHCPv4 relay...done.
Starting Unbound DNS...done.
Setting up gateway monitors...done.
Configuring firewall.......done.
Starting PFLOG...
pflog0: promiscuous mode disabled
done.
pflog0: promiscuous mode enabled
Syncing OpenVPN settings...done.
Starting NTP service...deferred.
Starting Unbound DNS...done.
Generating RRD graphs...done.
Configuring system logging...done.
>>> Invoking start script 'newwanip'
>>> Invoking start script 'freebsd'
Updating aliases
Certificates generated /usr/local/etc/postfix/cert_opn.pem
Certificates generated /usr/local/etc/postfix/ca_opn.pem
postfix/postfix-script: starting the Postfix mail system
[ALERT] 282/093839 (31656) : parsing [/usr/local/etc/haproxy.conf:102] : 'server anonymized_server_1' : could not resolve address 'anonymized_fqdn_1'.
[ALERT] 282/093839 (31656) : parsing [/usr/local/etc/haproxy.conf:103] : 'server anonymized_server_1' : could not resolve address 'anonymized_fqdn_2"
[ALERT] 282/093839 (31656) : Failed to initialize server(s) addr.
/usr/local/etc/rc.d/haproxy: WARNING: failed precmd routine for haproxy
>>> Invoking start script 'frr'
Checking zebra.conf
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: Disabling MPLS support (no kernel support)
OK
Starting zebra.
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: ifam_read() doesn't read all socket data
2019/10/10 09:43:52 warnings: ZEBRA: Disabling MPLS support (no kernel support)
Checking ripd.conf
OK
Starting ripd.
>>> Invoking start script 'vmware'
Starting vmware_guestd.
>>> Invoking start script 'carp'
>>> Invoking start script 'cron'
Starting Cron: OK
>>> Invoking start script 'beep'
Root file system: /dev/gpt/rootfs
`

**Comment by fraenki** [comment]
Two things I've spotted:

```
/usr/local/etc/rc.d/haproxy: WARNING: failed precmd routine for haproxy
```

As stated previously, HAProxy immediately fails to start when hostnames are unresolvable (instead of sitting there and waiting for something).

```
 postfix/postfix-script: starting the Postfix mail system
/usr/local/etc/rc.d/haproxy: WARNING: failed precmd routine for haproxy
Invoking start script 'frr'
```

Indeed, it looks like they are not started in the order that is shown by `rcorder`.
It looks like the postfix and HAProxy plugins use the expected startup order.
But frr is started by a syshook, which means that `rcorder` is ignored:

https://github.com/opnsense/core/blob/69139fcbb28d5080f1f7d32ad3c306cd7bd24310/src/etc/rc.syshook#L28-L42

Just a guess, I don't know anything about the frr plugin :)

**Comment by fichtner** [comment]
> But frr is started by a syshook, which means that rcorder is ignored:

But that means frr failed to start normally also as 50-frr is only a fix for when it doesn't start ;)

We could start haproxy later, e.g. 60-haproxy... I'd like to keep default plugin priority where it is and move things back so core stuff can use 20-49 and plugins can space out the other range.
